addon/model.js

```diff
@@ -331,12 +433,22 @@ export default class MegamorphicModel extends EmberObject {
   }
 
   get id() {
-    return this._internalModel.id;
+    if (false) {
+      if (!this._recordData) {
+        return null;
```

Comment:
This looks extremely suspect; let's see if we actually need this and if so add comments + test

Reply:
removed. Original problem was that the changing the ordering of assigning props in createArgs made id be set before _recordData, and Ember tries to get the prop before it sets it. I fixed the createArgs creation so this isn't a problem anymore